framework/src/actions/SetupMeshAction.C

```diff
@@ -152,6 +153,33 @@ SetupMeshAction::setupMesh(MooseMesh * mesh)
 void
 SetupMeshAction::act()
 {
+  if (_app.parameters().get<bool>("use_split"))
+  {
+    auto split_file = _app.parameters().get<std::string>("split_file");
+    if (_type != "FileMesh")
```

Comment:
I still don't think this is right. We have several other FileMesh derived types. I know you haven't instantiated the object here yet so using a virtual method or some other normal trick doesn't work but we might be able to make this all more robust with the old validParams trick (embedding a "base" type in validParams).